benchmarks/Main.hs

```diff
@@ -61,4 +62,6 @@ main = do
     , bench "findIndexR" $ whnf findIndexR ((<indexFindThreshold), as)
     , bench "findIndexR_naïve" $ whnf findIndexR_naive ((<indexFindThreshold), as)
     , bench "findIndexR_manual" $ whnf findIndexR_manual ((<indexFindThreshold), as)
+    , bench "minimumOn"  $ whnf (minOn (\x -> x*x*x)) as
+    , bench "maximumOn"  $ whnf (maxOn (\x -> x*x*x)) as
```

Comment:
I'd benchmark `U.minimumBy` and `U.maximumBy` directly, without intermediaries in another module.